### ISSUE TITLE
Attach lwt_seg6local program type

### ISSUE THREAD

**Issue by louisna** [libbpf/libbpf#247]
Hi,

First of all please excuse me to submit a new issue for this problem. I tried to contact you at XX but all my mails got rejected by the system. 

Until recently I worked with BCC to load programs into the kernel and communicate with them in the userspace. I jumped to libbpf for performance reasons, and libbpf seems to be the _future_ of bpf. 

I work with BPF_PROG_TYPE_LWT_SEG6LOCAL programs, that are attached to an IPv6 Segment Routing route with the associated End.BPF SRv6 action. After the program is loaded into the kernel via BCC, I could attach it to the right hook using pyroute2 - it requires the file descriptor of the program to attach it to the route.

However here with libbpf I cannot do the same. From the source code, I see that we cannot attach such programs:
https://github.com/libbpf/libbpf/blob/986962fade5dfa89c2890f3854eb040d2a64ab38/src/libbpf.c#L8653

https://github.com/libbpf/libbpf/blob/986962fade5dfa89c2890f3854eb040d2a64ab38/src/libbpf.c#L8537-L8538

It seems logical, as said earlier, such programs must be attached via an IPv6 Segment Routing route.
I tried to get the file descriptor of the program with `bpf_object_btf__fd` or `bpf_program__fd` but the given fd are wrong. For example, launching two times in parallel the same program (or even two different) give the same fd for both. 

Using `bpftool`, I can get the program id, but it seems to be insufficient to load the program into the kernel at the correct hook: giving the retrieved id to pyroute2 does not work. Trying to inject the `my_prog.bpf.o` object of my program with `iproute2` does not work either, as the program section I provide is not correct. Here is an example of a bpf program that will be compiled using libbpf:
```c
#include <linux/bpf.h>
#include <bpf/bpf_helpers.h>

char LICENCE[] SEC("license") = "Dual BSD/GPL";

SEC("lwt_seg6local")
int notify_ok(struct __sk_buff *skb) {
    bpf_printk("BPF triggered from packet with SRv6 !\n");
    
    return BPF_OK;
}
```

And the output of `iproute2` is:
```bash
Program section 'notify_ok' not found in ELF file!
Error fetching program/map!
Failed to parse eBPF program: Operation not permitted
```

So I wonder if I am doing something wrong here, or if it would be better to stick with BCC for now, as libbpf would be more suited for tracing purposes.

Once again, please excuse me to start an issue for this, but I do not have any other way of contact here.

Thank you in advance

**Comment by anakryiko** [comment]
Please see https://github.com/libbpf/libbpf#bpflibbpf-usage-and-questions and send email to bpf@vger.kernel.org. There are few BPF program types that have to be attached with iproute2 (not libbpf). But I hear rumors this might change soon. But again, please use bpf mailing list for such questions and discussions.
